### DISCUSSION TITLE
Modal overlay not covering the whole page

### DISCUSSION THREAD

**Discussion by mirjammalksoo** [twilio-labs/paste#1577] [Q&A] [answered]
Hey team! 

Wanted to let you know about an issue we currently see with Paste modal overlay not applying to the side/top navbars and only covering the page content. 

![image](https://user-images.githubusercontent.com/40227184/122410424-93faf280-cf8c-11eb-9dd7-30109fc2d71c.png)
![image](https://user-images.githubusercontent.com/40227184/122410530-abd27680-cf8c-11eb-8076-fd7e3bd0995b.png)

(Can be reproduced for example when accessing [https://www.stage.twilio.com/console/project/settings](https://www.stage.twilio.com/console/project/settings) with employee account and clicking "Close")

Changing the `z-index` to 2000 for `data-reach-dialog-overlay` inside reach-portal solves the issue inside legacy console and admin-center (this fix doesn't apply to modals inside iframe in Console Zen/New Console)

![image](https://user-images.githubusercontent.com/40227184/122410769-e3d9b980-cf8c-11eb-9a3b-1642667c912b.png)

I wondered if you are aware of such issue and if there could be a general fix inside Paste also :) Console team might be able to help with the solution for modal overlay inside iframe.

**Comment by TheSisb** [comment, marked as answer]
Hi @mirjammalksoo,

I believe this question has been asked before here: https://github.com/twilio-labs/paste/discussions/901.  Essentially, the `console` theme has a z-index fix for the old console pages, and for console Zen you can use the `default` theme.  There should be a flag to determine in which instance your modal is in.  

Since Paste is used in many different products, applying a blanket z-index change could be risky as Console is just one of many products. The issue seems to be that old console is using z-index values that aren't tokens.

Hope this helps and thanks for the detailed question!

**Reply by mirjammalksoo** [reply]
Got it! Thank you for the explanation!
